sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -190,4 +194,33 @@ object DataSourceUtils {
     case LegacyBehaviorPolicy.LEGACY => RebaseDateTime.rebaseGregorianToJulianMicros
     case LegacyBehaviorPolicy.CORRECTED => identity[Long]
   }
+
+  def generateDatasourceOptions(
+      extraOptionsMap: CaseInsensitiveStringMap, table: CatalogTable): Map[String, String] = {
+    val pathOption = table.storage.locationUri.map("path" -> CatalogUtils.URIToString(_))
+    val options = table.storage.properties ++ pathOption
+    if (!SQLConf.get.getConf(SQLConf.LEGACY_EXTRA_OPTIONS_BEHAVIOR)) {
+      // Check the same key with different values
+      table.storage.properties.foreach { case (k, v) =>
+        if (extraOptionsMap.containsKey(k) && extraOptionsMap.get(k) != v) {
+          throw new AnalysisException(
+            s"Fail to create datasource for the table ${table.identifier.table} since the table " +
```

Comment:
`create datasource` -> `resolve data source`

Reply:
`table.identifier.table`  -> `table.identifier`, so that we include database name as well.

Reply:
Copy, rephased in f3f8c8d.